File: go/src/pe002.go
package main

import (
	"fmt"
)

func main() {
	const MAX = 4000000
	var next, total int
	previous := 1
	current := 1
	for current < MAX {
		next = current + previous
		if current%2 == 0 {
			total += current
		}
		previous = current
		current = next
	}
	fmt.Printf("%d\n", total)
}

File: perl/pe002.pl
#!/usr/bin/perl -w
use strict;

my $total = 0;
my $max = 4000000;
my $previous = 1;
my $current = 1;
while ($current < $max) {
  my $next = $previous + $current;
  $total += $current if (!($current % 2));
  $previous = $current;
  $current = $next;
}
print "$total\n";
